src/lib.rs

```diff
@@ -808,6 +808,58 @@ impl Default for Box<CStr> {
     }
 }
 
+#[cfg(feature = "std")]
+use std::ffi::{CStr as StdCStr, CString as StdCString};
+
+#[cfg(feature = "std")]
+impl From<CString> for StdCString {
+    fn from(s: CString) -> StdCString {
+        unsafe { StdCString::from_raw(s.into_raw()) }
+    }
+}
+
+#[cfg(feature = "std")]
+impl<'a> From<&'a CStr> for &'a StdCStr {
+    fn from(s: &'a CStr) -> &'a StdCStr {
+        s.as_ref()
+    }
+}
+
+#[cfg(feature = "std")]
+impl From<StdCString> for CString {
+    fn from(s: StdCString) -> CString {
+        unsafe { CString::from_raw(s.into_raw()) }
+    }
+}
+
+#[cfg(feature = "std")]
+impl<'a> From<&'a StdCStr> for &'a CStr {
+    fn from(s: &'a StdCStr) -> &'a CStr {
+        unsafe { CStr::from_ptr(s.as_ptr()) }
```

Comment:
Prefer `from_bytes_with_nul_unchecked` and `to_bytes_with_nul`. This avoids needing to scan the string to find its length from the raw pointer.